packages/pwa/app/hooks/use-navigation.js

```diff
@@ -26,10 +29,21 @@ const useNavigation = () => {
          * @param  {...any} args - additional args passed to `.push` or `.replace`
          */
         (path, action = 'push', ...args) => {
-            const updatedHref = buildPathWithUrlConfig(path, {locale})
+            const locale = getLocaleFromSite(site, localeShortCode)
+
+            const defaultLocaleRefs = getLocaleRefsFromSite(site, site.l10n.defaultLocale)
+
+            const updatedHref = buildPathWithUrlConfig(
+                path,
+                {
+                    locale: locale.id || locale.alias,
+                    site: site.alias || site.id
+                },
+                {defaultLocaleRefs}
+            )
             history[action](path === '/' ? '/' : updatedHref, ...args)
         },
-        [locale]
+        [localeShortCode]
```

Comment:
Let's add `site` to this dependency array too?

Reply:
added it in